### ISSUE TITLE
Allow a Dynamic Authorisation Policy to be for unauthenticated users only

### ISSUE THREAD

**Issue by CrispinF** [cloudscribe/cloudscribe#1009]
At present it's possible to remove all requirements from a dynamic authorisation policy on the Edit page (/policymanagement/editpolicy?id=) which means that all users meet the policy, including anonymous users.
Sometimes it would be useful to have a policy that is ONLY met by unauthenticated users.
This could be achieved in the Edit page by adding a new checkbox similar to "Require Authenticated User" seen below, but "Require Unauthenticated User"

![image](https://github.com/user-attachments/assets/95630365-6ff8-414a-85c7-803bf5af1357)

There would be some potential complexity if we wanted a policy to be matched by users who are either unauthenticated OR are authenticated but only in certain roles, however I don't think we have any use cases for that so let's ignore it.

**Comment by JimKerslake** [comment]
A lot of the core concepts and implementation here are Microsoft's code, not cloudscribe's.
Really the whole mechanism for determining whether a user meets an authorisation policy, needs to be authenticated, is in a role etc. is all computed at MS level... so we'd need to delve into that and build out sideways with some extensions to core MS concepts I think.

**Comment by StewartBellamy** [comment]
Yes,  it doesn't come out the box like 'RequireRole' or 'RequireAuthenticatedUser'.  You'd probably have to code in an assertion like this:
`authBuilder.RequireAssertion(a => !a.User.IsAuthenticated());`

**Comment by JimKerslake** [comment]
Yes can see a couple of possible code paths to getting there (I was being even more low-level and contemplating building our own 
`OnlyAnonymousAuthorizationRequirement  : AuthorizationHandler<OnlyAnonymousAuthorizationRequirement>, IAuthorizationRequirement `

Seems tractable but we'd need to be sure it really did improve our lives sufficiently ;)
